Auth0/Telemetry.swift

```diff
@@ -69,20 +76,55 @@ public struct Telemetry {
         return items
     }
 
-    static func versionInformation(bundle: Bundle = Bundle(for: Credentials.classForCoder())) -> [String: String] {
+    static func versionInformation(bundle: Bundle = Bundle(for: Credentials.classForCoder())) -> [String: Any] {
         let version = bundle.infoDictionary?["CFBundleShortVersionString"] as? String ?? Telemetry.NoVersion
-        let dict = [
+        let dict: [String: Any] = [
             Telemetry.NameKey: Telemetry.LibraryName,
             Telemetry.VersionKey: version,
-            "swift-version": "3.0"
-            ]
+            Telemetry.EnvironmentKey: Telemetry.generateEnviroment()
+        ]
         return dict
     }
 
-    static func generateValue(fromInfo info: [String: String] = Telemetry.versionInformation()) -> String? {
+    static func generateEnviroment() -> [String: String] {
+        let platform = Telemetry.osInfo()
+        let env = [ "swift": Telemetry.swiftVersion(),
+                    "os": platform.0,
+                    "os_version": platform.1
+        ]
+        return env
+    }
+
+    static func generateValue(fromInfo info: [String: Any] = Telemetry.versionInformation()) -> String? {
         let data = try? JSONSerialization.data(withJSONObject: info, options: [])
         return data?.a0_encodeBase64URLSafe()
     }
+
+    static func swiftVersion() -> String {
```

Comment:
Why are you converting this? Seems like something you'll need to keep updated over time rather than just passing it as-is or splitting off and returning the major version number.